### PR TITLE
[Bug fix] Allow to disable authorize

### PR DESCRIPTION
**Pull request by vincentchalamon**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #465
| License       | MIT

### CODE REVIEW

**Comment by vincentchalamon** [comment]
Bug on Travis PHPUnit:
> PHP Fatal error:  Class 'PHPUnit\Framework\TestCase' not found in /home/travis/build/FriendsOfSymfony/FOSOAuthServerBundle/vendor/symfony/symfony/src/Symfony/Component/Form/Test/FormIntegrationTestCase.php on line 22

File: DependencyInjection/FOSOAuthServerExtension.php [hunk 1]
```diff
@@ -88,10 +88,12 @@ public function load(array $configs, ContainerBuilder $container)
             }
         }
 
-        if (!empty($config['authorize'])) {
-            $this->loadAuthorize($config['authorize'], $container, $loader);
+        if (empty($config['authorize'])) {
+            return;
         }
 
+        $this->loadAuthorize($config['authorize'], $container, $loader);
```
Comment (dinamic) [inline DependencyInjection/FOSOAuthServerExtension.php]: Isn't this change doing the exact same thing?

Comment (vincentchalamon) [reply DependencyInjection/FOSOAuthServerExtension.php]: It's not the same thing as my version exit the function, so lines 99->104 won't be executed. In original version, they are executed even if authorize feature is disabled, which is a bug IMO.
Still waiting for the team to review & merge this PR

Comment (dinamic) [reply DependencyInjection/FOSOAuthServerExtension.php]: Version 1.6 is about to be tagged soon. We may be able to squeeze this in.

I think you would have better chance of this getting thru if you solve the conflicts. Unit testing the change would be nice too.

**Comment by dinamic** [comment]
This seems to duplicate #462, but it's solution relies on returning early and I fancy that. Once this get's merged in, we should be able to close #462 and #465.

File: DependencyInjection/FOSOAuthServerExtension.php (hunk 1 above)
Comment (vincentchalamon) [reply DependencyInjection/FOSOAuthServerExtension.php]: Conflicts solved, but this class is not unit tested

Comment (dinamic) [reply DependencyInjection/FOSOAuthServerExtension.php]: You are right. I was mislead by the presence of the test file.

**Comment by dinamic** [comment]
@dkarlovi, could we have this in the milestone for 1.6?

**Comment by florianajir** [comment]
please merge it!

**Comment by dkarlovi** [comment]
Fixed in #554.

**Pull request closed without merging**